### DISCUSSION TITLE
v0.2.0 release for AnonCreds RS

### DISCUSSION THREAD

**Discussion by TimoGlastra** [hyperledger/anoncreds-rs#234] [General]
A discussion to discuss the v0.2.0 release for AnonCreds RS. Want to sync because there's quite some updates being made to the library itself and the wrappers.

Some questions from my side:
- Are the updates to the AnonCreds RS library to move it to the anoncreds-clsignatures library complete? (@petridishdev, @andrewwhitehead)
- Are the updates to make the library natively consumable complete (or at least ready for a new release)? (@bobozaur, @Patrik-Stas)
- Is the Python wrapper fully up to date? (@andrewwhitehead)
- Is the JS wrapper fully up to date? (@berendsliedrecht, @genaris)
  - note: we should add a version compatibility matrix to the js wrapper readmes
- Should we also add a release pipeline to crates for the rust library itself? (@andrewwhitehead, @bobozaur, @Patrik-Stas, @berendsliedrecht)

If all outstanding tasks are complete, I'd like to create a first 0.2.0-dev1 release so we can test out all the new changes before we release v0.2.0

Please respond here whether everything is done, or what's still left todo.

cc @swcurran

**Comment by bobozaur** [comment]
> * Are the updates to the AnonCreds RS library to move it to the anoncreds-clsignatures library complete? (@petridishdev, @andrewwhitehead)

I'm curious, what does this mean? Combining anoncreds-rs and anoncreds-clsignatures in a single library?
> * Are the updates to make the library natively consumable complete (or at least ready for a new release)? (@bobozaur, @Patrik-Stas)

#232 is pending, at least as far as a new release is concerned.
> * Should we also add a release pipeline to crates for the rust library itself? (@andrewwhitehead, @bobozaur, @Patrik-Stas, @berendsliedrecht)

Are you talking about releasing the Rust crates to `crates.io`?

**Comment by swcurran** [comment]
> I'm curious, what does this mean? Combining anoncreds-rs and anoncreds-clsignatures in a single library?

No — it means are we fully converted from using Ursa CL-Signatures to anoncreds-clsignatutes as a dependency.

**Comment by andrewwhitehead** [comment]
> * Are the updates to the AnonCreds RS library to move it to the anoncreds-clsignatures library complete? (@petridishdev, @andrewwhitehead)

Yes that's complete. I may have one more cleanup PR to take advantage of updates in version 0.2.2.

> * Is the Python wrapper fully up to date? (@andrewwhitehead)

Yes, I believe it's in good shape.

> * Should we also add a release pipeline to crates for the rust library itself? (@andrewwhitehead, @bobozaur, @Patrik-Stas, @berendsliedrecht)

That's probably a good idea.

**Comment by TimoGlastra** [comment]
We are aiming to create a new release for AFJ soon and with that want to include a new version of AnonCreds RS.

The current main branch is unstable and contains the new W3C code but without support for wrappers etc..

I've created a new branch in this repository v0.2.0, and pinned it to the last commit that we can release under v0.2.0 at the moment. If testing on our side gives no further issues with this version I want to release this as v0.2.0 and then we can release the code currently in main as v0.3.0 once that's ready.

Any objections to this approach?

The branch: https://github.com/hyperledger/anoncreds-rs/tree/v0.2.0

**Comment by swcurran** [comment]
@andrewwhitehead — can you weigh on this ASAP.

Particularly concerned with this:

> The current main branch is unstable and contains the new W3C code but without support for wrappers etc..

And the three W3C PRs that are still open.

**Comment by TimoGlastra** [comment]
v0.2.0 has been released: https://github.com/hyperledger/anoncreds-rs/releases/tag/v0.2.0
